### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by kazaneza**
**This pull request adds the following features to the final JavaScript project**:

 

- [ ] Implement Webpack for bundling and optimizing the code
- [ ]  Integrate API to fetch data from the server
- [ ]  Implement pagination to display movies
- [ ]  Update UI to make it more user-friendly and responsive

### **Here's a summary of the changes:**
 

- [ ] Added webpack configuration file
- [ ]  Added API integration to fetch data from the server using Fetch API
- [ ]  Updated UI with new styles and design elements
- [ ]  Added count and likes and comments

I've thoroughly tested these changes and ensured that they work correctly. Please let me know if you have any comments or feedback on this pull request.

Thank you!

### CODE REVIEW

File: src/modules/__test__/countShows.test.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  </ul>`;
+
+  it('should count the comment count', () => {
+    const element = document.createElement('div');
+    element.innerHTML = mazeGrid;
+
+    const numberOfShows = countShows(element);
+
+    expect(numberOfShows).toBe(3);
+  });
```
Comment (amejid) [inline src/modules/__test__/countShows.test.js:77]: - [ ] Great job with all the functionalities and organization of the code. 💯️ Let's add more test cases to cover different scenarios. One of the scenarios is when there are no displayed shows on the screen, the counter should return `0`. Kindly add a test for this situation. ✍️

File: src/modules/__test__/countComment.test.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+    // Act
+    openPopup(details, popupEl, null, targetEl);
+
+    // Assert
+    const commentCountEl = popupEl.querySelector('#comment-count');
+    const count = countComment(popupEl).toString();
+    expect(commentCountEl.textContent).toBe(count);
+  });
+
```
Comment (amejid) [inline src/modules/__test__/countComment.test.js:50]: - [ ] Could you add a test case for the scenario where no comments are displayed on the screen? We should cover edge cases as much as possible. 🤝️

File: src/modules/commentPopup.js [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+
+  // handle comment form submission
+  formEl.addEventListener('submit', async (e) => {
+    e.preventDefault();
+    const [username, comment, button] = e.target.elements;
+
+    if (!onCommentFormSubmit) return;
+
+    button.disabled = true;
+    button.innerHTML = 'Submitting...';
```
Comment (amejid) [inline src/modules/commentPopup.js:91]: - [ ] I like how you are disabling the button momentarily when a comment is submitted. Good job. :+1:

**Review by amejid — CHANGES_REQUESTED** [review]
Hi @kazaneza & @Nezerwa, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
- Linter checks passing :heavy_check_mark: 
- All functionalities working as expected :heavy_check_mark: 
- Highly organized code :fire: :fire: 
- Good UI design :fire: :fire: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Olamarx — APPROVED** [review]
Hi @kazaneza, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Linters passed ✔️ 
- Tests passed ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
